crates/rust-client/src/notes/import.rs

```diff
@@ -35,6 +35,18 @@ impl<R: FeltRng> Client<R> {
 
         let previous_note = self.get_input_note(id).await.ok();
 
+        // If the note is already in the store and is in the state processing we return an error.
+        if let Some(
+            InputNoteState::ProcessingAuthenticated(_)
+            | InputNoteState::ProcessingUnauthenticated(_),
+        ) = previous_note.as_ref().map(|note| note.state())
+        {
+            return Err(ClientError::NoteImportError(format!(
+                "Note with id {} is already being processed",
```

Comment:
I don't think the "already" makes much sense here as we are not trying to process it again. We are importing and overwriting the old note. Maybe something like:
```suggestion
                "Can't overwrite note with id {} as it's currently being processed",
```